### PR TITLE
Enable autocomplete

### LINKED ISSUE #697: Add option to auto-complete PRs in VS to the insertion tool
Would be nice to have an option to allow a merge from GitHub PR to automatically trigger everything including an automatic insertion

### PR DESCRIPTION
**Pull request by dmonroym**
Fixes #697

### CODE REVIEW

File: src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs [hunk 1]
```diff
@@ -198,6 +198,56 @@ private static async Task QueueBuildPolicy(GitPullRequest pullRequest, string bu
...
+                        }
+                    },
+                    repository.Id,
+                    pullRequest.PullRequestId
+                    );
+                Console.WriteLine($"Updated {pullRequest.Description} with AutoComplete");
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine($"Could not set AutoComplete: {e.GetType().Name} : {e.Message}");
```
Comment (dmonroym) [inline src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs]: Not sure if I should throw here. Don't want the release to fail if we couldn't set the PR to auto-complete but also want to inform the user - is it best to throw or not?

File: src/RoslynInsertionTool/scripts/OneOffInsertion.ps1 [hunk 2]
```diff
@@ -16,7 +16,8 @@ param([string] $clientId,
       [string] $visualStudioBranchName,
       [string] $titlePrefix,
       [string] $writePullRequest,
-      [int] $insertionCount)
+      [int] $insertionCount,
+      [string] $autoComplete)
```
Comment (dmonroym) [inline src/RoslynInsertionTool/scripts/OneOffInsertion.ps1]: Not super familiar with PS but would adding this param break any existing uses of this script or is it optional?

**Comment by dmonroym** [comment]
Not entirely sure who the right reviewers are for this but please add them if you know who they are

File: src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs (hunk 1 above)
Comment (jmarolf) [reply src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs]: WE typically logged things here. I think all that code was removed. I would at least write to the console window so people know what happened

---
In reply to: [377406314](https://github.com/dotnet/roslyn-tools/pull/698#discussion_r377406314) [](ancestors = 377406314)

File: src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs [hunk 3]
```diff
@@ -198,6 +198,56 @@ private static async Task QueueBuildPolicy(GitPullRequest pullRequest, string bu
             }
         }
 
+        /// <summary>
+        /// There is no enum or class in Microsoft.TeamFoundation.SourceControl.WebApi defined for vote values so made my own here.
+        /// Values are documented at https://www.visualstudio.com/en-us/docs/integrate/api/git/pull-requests/reviewers#update-a-reviewers-vote.
```
Comment (RikkiGibson) [inline src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs]: This URL redirects me to the ADO Git REST API docs. https://docs.microsoft.com/en-us/rest/api/azure/devops/git/?view=azure-devops-rest-5.0#update-a-reviewers-vote

File: src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.cs [hunk 4]
```diff
@@ -19,7 +19,10 @@ namespace Roslyn.Insertion
     public static partial class RoslynInsertionTool
     {
         public static readonly Guid VSRepoId = new Guid("a290117c-5a8a-40f7-bc2c-f14dbe3acf6d");
+        //Easiest way to get these GUIDs is to create a PR search in AzDo
```
Comment (RikkiGibson) [inline src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.cs]: Thanks for adding this note.

File: src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs (hunk 1 above)
Comment (RikkiGibson) [reply src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs]: There should be a LogWarning method you can use which will make it show up as a warning in the release log. e.g. https://devdiv.visualstudio.com/DevDiv/_releaseProgress?_a=release-environment-logs&releaseId=580152&environmentId=3097439

File: src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.cs [hunk 5]
```diff
@@ -378,6 +381,22 @@ public static partial class RoslynInsertionTool
...
                 }
 
+                // ********************* Set PR to Auto-Complete *****************************
+                if (Options.SetAutoComplete)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    Console.WriteLine($"Set PR to Auto-Complete");
+                    try
+                    {
+                        await SetAutoCompleteAsync(pullRequest);
```
Comment (RikkiGibson) [inline src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.cs]: should this method take a CancellationToken?

File: src/RoslynInsertionTool/scripts/OneOffInsertion.ps1 (hunk 2 above)
Comment (RikkiGibson) [reply src/RoslynInsertionTool/scripts/OneOffInsertion.ps1]: I know basically nothing about powershell. I think if you are able to run `.\src\RoslynInsertionTool\scripts\VS-Insertion.ps1` without having to add an argument to OneOffInsertion cmdlet call we are good. @jmarolf?

File: src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs (hunk 3 above)
Comment (dmonroym) [reply src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs]: Updated to https://docs.microsoft.com/en-us/dotnet/api/microsoft.teamfoundation.sourcecontrol.webapi.identityrefwithvote.vote?view=azure-devops-dotnet

File: src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.cs (hunk 5 above)
Comment (dmonroym) [reply src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.cs]: I'm mimicking the behavior of QueueValidationBuild, so yes?

Comment (RikkiGibson) [reply src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.cs]: Just to be clear I'm referring to `SetAutoCompleteAsync`. It doesn't meaningfully affect things to not be able to cancel here but feels like we should be consistent. If it works as is then no big deal.

File: src/RoslynInsertionTool/scripts/OneOffInsertion.ps1 (hunk 2 above)
Comment (dmonroym) [reply src/RoslynInsertionTool/scripts/OneOffInsertion.ps1]: Just tested this out and if no value is passed it defaults autoComplete=false

File: src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.cs (hunk 5 above)
Comment (dmonroym) [reply src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.cs]: oh, I guess I could easily add it there. To keep conistency though, should I add it to the QueueBuildPolicy too?

File: src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs [hunk 6]
```diff
@@ -222,7 +222,8 @@ private static async Task SetAutoCompleteAsync(GitPullRequest pullRequest)
                     new IdentityRefWithVote { Vote = (short)Vote.Approved },
                     repository.Id,
                     pullRequest.PullRequestId,
-                    VSLSnapUserId.ToString()
+                    VSLSnapUserId.ToString(),
+                    cancellationToken : cancellationToken
```
Comment (RikkiGibson) [inline src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs]: nit: formatting (no rush, just would be good to fix up in a future PR)

Comment (dmonroym) [reply src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs]: what's the right format?

**Pull request merged**